samples/monitor-changes-to-draw-status/src/main/java/com/esri/arcgismaps/sample/monitorchangestodrawstatus/screens/MonitorDrawStatusScreen.kt

```diff
@@ -0,0 +1,119 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.monitorchangestodrawstatus.screens
+
+import android.content.res.Configuration
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.arcgismaps.mapping.view.DrawStatus
+import com.arcgismaps.toolkit.geoviewcompose.MapView
+import com.esri.arcgismaps.sample.monitorchangestodrawstatus.components.MonitorDrawStatusViewModel
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialog
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+import com.esri.arcgismaps.sample.sampleslib.components.SamplePreviewSurface
+
+/**
+ * Main screen for the Monitor changes to draw status sample.
+ *
+ * The composable observes the ViewModel's mapIsDrawing flow and updates the UI
+ * when the draw status changes. The MapView's onDrawStatusChanged callback
+ * is used to notify the ViewModel of draw status changes.
+ */
+@Composable
+fun MonitorDrawStatusScreen(sampleName: String) {
+    val viewModel: MonitorDrawStatusViewModel = viewModel()
+
+    // Observe whether the map is currently drawing
+    val mapIsDrawing by viewModel.mapIsDrawing.collectAsStateWithLifecycle(false)
+
+    Scaffold(
+        topBar = { SampleTopAppBar(title = sampleName) },
+        content = { padding ->
+            Column(modifier = Modifier
+                .fillMaxSize()
+                .padding(padding)) {
+
+                Box(modifier = Modifier.fillMaxSize()) {
+                    // Compose MapView: pass in the ArcGISMap from the ViewModel and the draw status callback.
+                    MapView(
+                        modifier = Modifier
+                            .fillMaxSize(),
+                        arcGISMap = viewModel.arcGISMap,
+                        onDrawStatusChanged = { drawStatus: DrawStatus ->
+                            // Forward draw status updates to the ViewModel
+                            viewModel.updateDrawStatus(drawStatus)
+                        }
```

Comment:
Could be simplified: 

```suggestion
                        // Forward draw status updates to the ViewModel
                        onDrawStatusChanged = viewModel::updateDrawStatus
```